src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Tests/ConfigurationMockTests.cs

```diff
@@ -257,6 +257,76 @@ public async Task GetBatch()
             AssertRequestCommon(request1);
         }
 
+        [Test]
+        public async Task GetBatchWithfilters()
+        {
+            var selector = new SettingSelector()
+            {
+                Fields = SettingFields.Key | SettingFields.Label | SettingFields.ETag
+            };
+
+            var response1 = new MockResponse(200);
+            response1.SetContent(SerializationHelpers.Serialize(new[]
+            {
+                CreateResponseSetting(0, selector),
+                CreateResponseSetting(1, selector),
+            }, SerializeBatch));
+
+            var mockTransport = new MockTransport(response1);
+            ConfigurationClient service = CreateTestService(mockTransport);
+
+            int keyIndex = 0;
+            using (Response<SettingBatch> response = await service.GetBatchAsync(selector, CancellationToken.None))
+            {
+                SettingBatch batch = response.Value;
+                for (int i = 0; i < batch.Count; i++)
+                {
+                    ConfigurationSetting value = batch[i];
+                    Assert.AreEqual("key" + keyIndex, value.Key);
+                    Assert.IsNotNull(value.Label);
+                    Assert.IsNotNull(value.ETag);
+                    Assert.IsNull(value.Value);
+                    keyIndex++;
+                }
+            }
+        }
+
+        [Test]
+        public async Task GetBatchWithAllfilters()
+        {
+            var selector = new SettingSelector()
+            {
+                Fields = SettingFields.All
+            };
+
+            var response1 = new MockResponse(200);
+            response1.SetContent(SerializationHelpers.Serialize(new[]
+            {
+                CreateResponseSetting(0, selector),
+                CreateResponseSetting(1, selector),
+            }, SerializeBatch));
+
+            var mockTransport = new MockTransport(response1);
+            ConfigurationClient service = CreateTestService(mockTransport);
+
+            int keyIndex = 0;
+            using (Response<SettingBatch> response = await service.GetBatchAsync(selector, CancellationToken.None))
+            {
+                SettingBatch batch = response.Value;
+                for (int i = 0; i < batch.Count; i++)
+                {
+                    ConfigurationSetting value = batch[i];
+                    Assert.AreEqual("key" + keyIndex, value.Key);
+                    Assert.IsNotNull(value.Label);
```

Comment:
I think this only tests test code itself. Asserting URIs should be enough for Mock tests.

Reply:
True... while writing it, it felt more in parity with the behavior the live service expects, but it is true that because everything is a mock (even serialization of the response) there isn't much value.

Reply:
If we think about Asserting the URIs, then the `ConfigurationSettingTests.SettingAllFields` and `SettingSomeFields` should do the trick.